src/Estimators/src/SubModel.cpp

```diff
@@ -371,27 +390,29 @@ bool blfEstim::SubModelCreator::createSubModels(
     std::vector<std::string> gyroNames, gyroFrames;
     ok = ok && populateSensorParameters("GYROSCOPE", gyroNames, gyroFrames);
 
-    std::vector<blfEstim::Sensor> gyroList;
+    std::vector<RDE::Sensor> gyroList;
     for (auto idx = 0; idx < gyroNames.size(); idx++)
     {
-        blfEstim::Sensor gyro;
+        RDE::Sensor gyro;
         gyro.name = gyroNames[idx];
         gyro.frame = gyroFrames[idx];
         gyroList.push_back(std::move(gyro));
     }
 
     ok = ok
-         && blfEstim::SubModelCreator::createSubModels(ftList, accList, gyroList, extContactFrames);
+         && RDE::SubModelCreator::createSubModels(ftList, accList, gyroList, extContactFrames);
 
     return ok;
 }
 
-void blfEstim::SubModelCreator::setModel(const iDynTree::Model& model)
+void RDE::SubModelCreator::setModelAndSensors(iDynTree::ModelLoader& modelLoader)
 {
-    this->m_model = model;
+    m_model = modelLoader.model();
+
+    m_sensorList = modelLoader.sensors();
 }
```

Comment:
From outside, it can be a bit misleading the fact that the method is ``setModelAndSensors``, but you are passing a single input. Moreover, in this way you limit yourself to the use of ``iDynTree::ModelLoader``, but that is not strictly necessary. I would suggest to change the signature to 
```cpp
void setModelAndSensors(const iDynTree::Model& mode, iDynTree::SensorsList& sensors);
```
and from outside you will simply call it with
```cpp
subModel.setModelAndSensors(mdldr.model(), mdldr.sensors())
```

Reply:
Done!